draft-normington-jsonpath-latest.xml

```diff
@@ -381,57 +381,66 @@ array-slice = [ integer ] ws %x3A ws [ integer ]
-                        The index <tt>0</tt> selects the first element of a non-empty array and, providing <tt>i &lt; len - 1</tt>, index <tt>i + 1</tt>
-                        selects the element immedately after (in array order) the element selected by index <tt>i</tt>.
-                     </t>
-                     <t>A negative index <tt>j</tt> selects an element of an array of length <tt>len</tt> if and only if <tt>0 &lt;= j + len &lt; len</tt>,
-                        in which case it selects the same element as the non-negative index <tt>j + len</tt>.
-                     </t>
-                     <t>An array slice selects any elements of an array which are indexed by the members of the slice,
-                        as defined below, in the same order as the slice.
-                        It selects no nodes from a node which is not an array.
-                     </t>
-                     <t>The members of a slice depend on whether the step value is positive, negative, or zero as follows. If the step value is
-                        omitted, it defaults to <tt>1</tt>.
-                     </t>
-                     <t>For positive integer <tt>step</tt> and integers <tt>start</tt> and <tt>end</tt>, the array slice
-                        <tt>start:end:step</tt> consists of the indices starting at <tt>start</tt> and increasing repeatedly
-                        by <tt>step</tt> while the resultant index is less than <tt>end</tt>.
-                        The indices of the slice are the same as the values of i iterated over by this <tt>for</tt> loop in C:
-                     </t>
+                     <t>A slice expression produces a new array containing a subset of the elements of the input array.</t>
+                     <t>A slice is defined by the two bounds, <tt>start</tt> and <tt>end</tt>,
+                        and a <tt>step</tt> which governs the direction of iteration and the iteration stride.
+                        Each of those parameters is optional.</t>
+                     <t>Negative bounds are interpreted as offsets from the end of the array, namely <tt>b + len</tt>.
+                        This remapping is performed before any other evaluation step.</t>
+                     <t>A slice bound <tt>b</tt> is considered valid if <tt>0 &lt;= b &lt; len</tt>.
+                        An invalid bound is treated as an omitted bound. Omitted bounds take a default value
+                        that depends on the value of the <tt>step</tt> parameter:</t>
+                     <t>If <tt>step &gt; 0</tt>, then <tt>start</tt> defaults to <tt>0</tt>
+                        while <tt>end</tt> defaults to <tt>len</tt>.</t>
+                     <t>If <tt>step &lt; 0</tt>, then <tt>start</tt> defaults to <tt>len-1</tt>,
+                        while an omitted <tt>end</tt> causes the iteration to terminate at 0 (inclusive).</t>
```

Comment:
So the earlier statement of omitted bounds taking a default value isn't _quite_ accurate as in this case, we are avoiding saying what the default value is.

Reply:
PTAL